### ISSUE TITLE
Displayed CPU/GPU count for app is not updated when app_config.xml changes them

### ISSUE THREAD

**Issue by sirzooro** [BOINC/boinc#1903]
When you change number of required CPUs and/or GPUs for app using app_config.xml, GUI is not updated - it still shows old values. 3rd party software like BoincTasks also shows it in this way, so BOINC Client sends them incorrect values. This is display-only issue, number of running tasks is properly adjusted. New tasks downloaded later shows correct value. If I remember correctly, BOINC restart also fixes this.

**Comment by Ageless93** [comment]
You're adding values of how many tasks can be run at the same time through
a configuration file, not through a preferences file. BOINC reads those
values from the global_prefs.xml or global_prefs_override.xml files which
aren't changed when the user uses an app_config.xml or app_info.xml file.

Additionally, you're not changing the CPU/GPU count via an app_config.xml
file, you're just telling BOINC how many tasks it can run at the same time
via this file.

I think you mean when using the avg_ncpus and ngpus value?
Those just tell BOINC how many instances of each the application can use,
it doesn't tell BOINC that those are the only values it can use all over.
That's set via the preferences.


-- Jord van der Elst.

**Comment by CharlieFenton** [comment]
Did you remember to select "Read config files" from the options menu in the GUI after modifying app_config.xml?

**Comment by sirzooro** [comment]
I modified the `<cpu_usage>/<gpu_usage>` or `<avg_ncpus>/<max_ncpus>` (I did this for different apps) in app_config.xml - for all of them results is the same. I did not use `<max_concurrent>` tag there, so BOINC determined how may app instances can start.

I reloaded config files using "Read config files" option after doing my changes.

I did not change global preferences, they are set to use 100% of available CPUs. My cc_config.xml also does not have any tags which changes this.

Problem is with CPU/GPU count displayed by BOINC Mgc for some apps in parenthesis after app status, or by BoincTasks in Usage column. This is display problem, internally BOINC uses correct values.
